controllers/gitopsservice_controller.go

```diff
@@ -472,9 +472,15 @@ func (r *ReconcileGitopsService) reconcileDefaultArgoCDInstance(instance *pipeli
 			changed = true
 		}
 
-		if !reflect.DeepEqual(existingArgoCD.Spec.NodePlacement, defaultArgoCDInstance.Spec.NodePlacement) {
-			existingArgoCD.Spec.NodePlacement = defaultArgoCDInstance.Spec.NodePlacement
-			changed = true
+		// Checking if default nodePlacement is nil to verify whether the patch is done using GitopService CR or ArgoCD CR
+		// if user is patching nodePlacement through GitopsService CR, then the above written logic for GitopsService will add custom values to defaultArgoCDInstance's nodePlacement,
```

Comment:
The logic is that unless any change is made in GitopService CR, don't check the nodePlacement spec right?
That will allow users to manage the default instance via ArgoCD CR.

The comment is confusing and not sure if needed.

Reply:
In the bug, user is trying to patch nodePlacement using `oc patch argocd ...` and logic present is checking for nodePlacement when user tries to patch using gitopsservice `oc patch gitopsservice ...` . Basically in comment it is mentioned that if the user is trying to patch argocd using gitopsservice then `defaultArgoCDInstance.Spec.NodePlacement` will not be nil and it will update argocd(i.e, existingArgoCD.Spec.NodePlacement ) based on that, otherwise `defaultArgoCDInstance.Spec.NodePlacement` will be nil and it will not update the `existingArgoCD.Spec.NodePlacement ` now as I have added If condition here to check this, I have added the explanation in the description, please let me know if its not making sense

Reply:
IIUC, what Saumeya is trying to convey is, instead of this comment i.e. explaining how this part of code works, if we can just reduce the comment to what it does. 
Maybe we can just write 

```
     // if user is patching nodePlacement through GitopsService CR, then existingAgoCd NodePlacement is updated.
```

Reply:
got it, I will update this, thanks Saumeya and Anand